tasks/Functions-admin/upgrade.js

```diff
@@ -0,0 +1,60 @@
+const { networkConfig } = require("../../network-config")
+const { types } = require("hardhat/config")
+
+task("functions-upgrade", "Upgrades the implementation of an existing Functions contract")
+  .addParam("type", "Name of the contract to upgrade, options: oracle, registry", null, types.string)
+  .setAction(async (taskArgs) => {
+    console.log("\n__Recompiling Contracts__")
+    await run("compile")
+
+    if (network.name === "hardhat") {
+      throw Error("This command cannot be used on a local development chain.  Specify a valid network.")
+    }
+
+    let proxyAddress, newImplementationFactory, contractPath
+    const { type } = taskArgs
+    if (type === "registry" || type === "Registry" || type === "REGISTRY") {
```

Comment:
How about 
`if (type.toLowerCase() === "registry")`

Reply:
That does sound cleaner. Changing.